_posts/en/2019-02-04-rbf-in-the-wild.md

```diff
@@ -0,0 +1,408 @@
+findings were initially presented during the most recent [Bitcoin Optech
+Workshop in Paris](/workshops/).
+
+Understanding how other wallets, exchanges, and block explorers handle opt-in
+RBF transactions is an important consideration for Bitcoin services. In order
+for exchanges or wallets to support RBF, they want to know that there is broad
+support for RBF and that user experience in receiving wallets or block
+explorers is not confusing.
+
+The goal of this study is to evaluate how opt-in RBF is supported in the wallet
+and explorer ecosystem. While the study is meant to be more descriptive than
+prescriptive, we do have some general recommendations. When evaluating
+usability, each service should consider the experience and knowledge of its
+user base. Hopefully, the compilation of this information in one location will
+help Bitcoin companies make informed decisions about how to support RBF and
+even start some discussions about which techniques work and which do not.
+
+Transaction replacement is important as it can enable:
+
+- Wallets and users to have finer control over transaction fees since an
+  initially low fee can later be bumped
+- 'Unsticking' a stuck transaction that doesn't have enough fee to be mined
+
+As well as those basic use cases, RBF can be used in more advanced scenarios
+such as:
+
+- Combining two or more payments into a single transaction (Iterative Payment
+  Batching)
+- Pre-computed fee bumping
+- Closing long-lasting payment channels where the fees cannot be predicted
+  ahead of time.[^closing_lightning_channels]
+
+## Background
+
+[Full transaction
+replacement](https://en.bitcoin.it/wiki/Replace_by_fee#Full_RBF) was available
+from Bitcoin’s original 0.1.0 release until 0.3.12, but there were a few
+potential downsides. Namely, the unrestricted ability to replace transactions
+meant there was no rate limiting mechanism to prevent DoS/spam replacements.
+There was also no incentive for a miner to replace the transaction, especially
+if the original transaction had the same or higher feerate.
+
+Satoshi [disabled transaction
+replacement](https://github.com/bitcoin/bitcoin/commit/05454818dc7ed92f577a1a1ef6798049f17a52e7#diff-118fcbaaba162ba17933c7893247df3aR522)
+in 0.3.12, leaving the comment _“// Disable replacement feature for now”_. From
+0.3.12 to 0.11.x Bitcoin Core had no default transaction replacement
+capabilities.
+
+That changed when an implementation of Peter Todd’s [BIP125 “Opt-in Full
+Replace-by-Fee
+Signaling”](https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki) was
+merged into Bitcoin Core 0.12.0. BIP125 solves both the spam prevention and
+incentive issues mentioned above by requiring higher transaction fees for
+replacement transactions.
+
+![rbf-transactions-in-2018](/img/posts/rbf-in-the-wild/rbf-transactions-in-2018.png)
+*~6% of transactions signaled opt-in RBF in 2018. Source: [Bitcoin Optech
```

Comment:
In my fixup commit: I've changed this to ~6% and signal -> signaled now that we're in 2019.